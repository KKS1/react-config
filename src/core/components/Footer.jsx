import React, { Component } from "react";
import Utils from "../../core/helpers/utils";
import _ from "lodash";

//Internal utility method to generate button markup
const footerButtonTemplate = buttonConfig => {
  const {
    text,
    handler,
    classes,
    buttonClasses,
    iconClasses,
    hasPreTextIcon
  } = buttonConfig;
  const NavIcon = iconClasses ? (
    <div className={iconClasses} aria-hidden="true" />
  ) : null;
  const key = _.uniqueId();
  const className = `${classes} form-group`;
  const buttonClassName = `btn btn--block btn--ghost js-substep-nav ${buttonClasses}`;

  return (
    <div className={className} key={key}>
      <button type="button" className={buttonClassName} onClick={handler}>
        {hasPreTextIcon && NavIcon}
        {text}
        {!hasPreTextIcon && NavIcon}
      </button>
    </div>
  );
};

// Generates template for application release footer
const releaseTemplate = props => {
  const releaseButtonConfig = {
    text: "Save and return to review",
    handler: props.onBackToReview,
    classes: "col-12 col-md-4"
  };
  return footerButtonTemplate(releaseButtonConfig);
};

// Generates template for the default footer with back and next buttons
const defaultTemplate = props => {
  const defaultIconClasses = "btn__icon btn__icon--lg";
  const defaultButtonsConfig = {
    next: {
      text: "Next",
      handler: props.onNextClick,
      classes: "col-6 col-md-2",
      buttonClasses: "js-form-footer-next",
      iconClasses: `${defaultIconClasses} btn__icon--pad-left far fa-chevron-right c-icon`
    },
    back: {
      text: "Back",
      hasPreTextIcon: true,
      handler: props.onBackClick,
      classes: "col-6 col-md-2",
      buttonClasses: "js-form-footer-back",
      iconClasses: `${defaultIconClasses} btn__icon--pad-right far fa-chevron-left c-icon`
    }
  };
  return [
    footerButtonTemplate(defaultButtonsConfig.next),
    footerButtonTemplate(defaultButtonsConfig.back)
  ];
};

export default class Footer extends Component {
  nextClick() {
    this.props.onNextClick();
  }

  backClick() {
    this.props.onBackClick();
  }

  static defaultProps = {
    onNextClick: () => {},
    onBackClick: () => {}
  };

  render() {
    const template = Utils.isRedirectedFromRelease()
      ? releaseTemplate(this.props)
      : defaultTemplate(this.props);
    return (
      <div className="c-loan-application__footer component row">{template}</div>
    );
  }
}
