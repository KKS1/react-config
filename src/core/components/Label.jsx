import React, { Component } from "react";

const Label = ({ schema, label, id }) => {
  return !schema && label ? <label htmlFor={id}>{label}</label> : "";
};

export default Label;
