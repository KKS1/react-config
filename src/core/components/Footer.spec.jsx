import React, { Component } from "react";
import Label from "./Label";
import Footer from "./Footer";

describe("Footer", () => {
  test("render_Footer", () => {
    const wrapper = shallow(<Footer />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });

  test("call_back_and_next_handler_props", () => {
    const backClickHandler = jest.fn();
    const nextClickHandler = jest.fn();

    const props = {
      id: "formOne_labelOne",
      value: 100,
      onBackClick: backClickHandler,
      onNextClick: nextClickHandler
    };
    const wrapper = shallow(<Footer {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();

    wrapper.find(".js-form-footer-back").simulate("click");

    expect(backClickHandler).toHaveBeenCalled();

    wrapper.find(".js-form-footer-next").simulate("click");

    expect(nextClickHandler).toHaveBeenCalled();
  });
});
