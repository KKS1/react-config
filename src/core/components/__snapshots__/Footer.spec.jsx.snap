// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer call_back_and_next_handler_props 1`] = `
Footer {
  "context": Object {},
  "props": Object {
    "id": "formOne_labelOne",
    "onBackClick": [MockFunction],
    "onNextClick": [MockFunction],
    "value": 100,
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <Footer
        id="formOne_labelOne"
        onBackClick={[MockFunction]}
        onNextClick={[MockFunction]}
        value={100}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-loan-application__footer component row"
      >
        <div
          className="col-6 col-md-2 form-group"
        >
          <button
            className="btn btn--block btn--ghost js-substep-nav js-form-footer-next"
            onClick={[MockFunction]}
            type="button"
          >
            Next
            <div
              aria-hidden="true"
              className="btn__icon btn__icon--lg btn__icon--pad-left far fa-chevron-right c-icon"
            />
          </button>
        </div>
        <div
          className="col-6 col-md-2 form-group"
        >
          <button
            className="btn btn--block btn--ghost js-substep-nav js-form-footer-back"
            onClick={[MockFunction]}
            type="button"
          >
            <div
              aria-hidden="true"
              className="btn__icon btn__icon--lg btn__icon--pad-right far fa-chevron-left c-icon"
            />
            Back
          </button>
        </div>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`Footer render_Footer 1`] = `
Footer {
  "context": Object {},
  "props": Object {
    "onBackClick": [Function],
    "onNextClick": [Function],
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <Footer
        onBackClick={[Function]}
        onNextClick={[Function]}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-loan-application__footer component row"
      >
        <div
          className="col-6 col-md-2 form-group"
        >
          <button
            className="btn btn--block btn--ghost js-substep-nav js-form-footer-next"
            onClick={[Function]}
            type="button"
          >
            Next
            <div
              aria-hidden="true"
              className="btn__icon btn__icon--lg btn__icon--pad-left far fa-chevron-right c-icon"
            />
          </button>
        </div>
        <div
          className="col-6 col-md-2 form-group"
        >
          <button
            className="btn btn--block btn--ghost js-substep-nav js-form-footer-back"
            onClick={[Function]}
            type="button"
          >
            <div
              aria-hidden="true"
              className="btn__icon btn__icon--lg btn__icon--pad-right far fa-chevron-left c-icon"
            />
            Back
          </button>
        </div>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
